end_to_end/end_to_end_suite_test.go

```diff
@@ -2242,6 +2242,71 @@ LANGUAGE plpgsql NO SQL;`)
 				"tplain": 1000, "ths": 1000, "tht": 1000})
 		})
 	})
+	Describe("Properly hanldes user defined type if different schema used", func() {
+		It("restores table with enum when exporting statistics", func() {
+			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
+													create type cdm.status AS ENUM ('ACTIVATING', 'ACTIVATED', 'ERROR');
+													create cast (text as cdm.status) with inout as implicit;`)
+			testhelper.AssertQueryRuns(backupConn, `create table cdm.contact_status (
+															contact_id uuid not null,
+															status cdm.status not null
+														) distributed by (contact_id);
+													insert into cdm.contact_status 
+													select 
+													md5(random()::text)::uuid,
+													(array['ACTIVATING', 'ACTIVATED', 'ERROR'])[1 + x%3]
+													from generate_series(1,20000) x;
+													analyze cdm.contact_status;
+												`)
+
+			defer func() {
+				testhelper.AssertQueryRuns(backupConn, `drop table cdm.contact_status;
+														drop cast (text as cdm.status);
+														drop type cdm.status cascade;
+														drop schema cdm cascade;`)
+				testhelper.AssertQueryRuns(restoreConn, `drop table cdm.contact_status;
+														drop cast (text as cdm.status);
+														drop type cdm.status cascade;
+														drop schema cdm cascade;`)
+			}()
```

Comment:
how about some simplification
```suggestion
			testhelper.AssertQueryRuns(backupConn, `create schema cdm;
													create type cdm.status as enum ('ACTIVATING', 'ACTIVATED', 'ERROR');
													create table cdm.contact_status (
														contact_id uuid not null,
														status cdm.status not null
													) distributed by (contact_id);
													insert into cdm.contact_status select
														md5(random()::text)::uuid,
														(array['ACTIVATING', 'ACTIVATED', 'ERROR'])[1 + x%3]::cdm.status
													from generate_series(1, 20000) x;`)

			defer func() {
				testhelper.AssertQueryRuns(backupConn, `drop schema cdm cascade;`)
				testhelper.AssertQueryRuns(restoreConn, `drop schema cdm cascade;`)
			}()
```